lib/rubyXL/objects/ooxml_object.rb

```diff
@@ -246,10 +304,27 @@ def preserve_whitespace
     end
     private :preserve_whitespace
 
+    # Manually update a hashing part.
+    def hash_dirty!
+      @need_hash_recompute = true
+    end
+
+    # Recompute the hash with segment given by the childrens
+    def recompute_hash
+      @hash = 0
+      @hash = obtain_class_variable(:@@ooxml_attributes).inject(@hash) { |h, (k, v)| h ^ ( v[:accessor].hash * self.send(v[:accessor]).hash ) }
+      @hash = obtain_class_variable(:@@ooxml_child_nodes).inject(@hash) { |h, (k, v)| h ^ ( v[:accessor].hash * self.send(v[:accessor]).hash )  }
+      @need_hash_recompute = false
+    end
+
+    def hash
+      recompute_hash if @need_hash_recompute
+      @hash
+    end
+
     def ==(other)
       other.is_a?(self.class) &&
-        obtain_class_variable(:@@ooxml_attributes).all? { |k, v| self.send(v[:accessor]) == other.send(v[:accessor]) } &&
-        obtain_class_variable(:@@ooxml_child_nodes).all? { |k, v| self.send(v[:accessor]) == other.send(v[:accessor]) }
+        self.hash == other.hash
```

Comment:
Just because the hashes are equal does not mean the attributes are equal.

You will still need to do the comparison of attributes when the hashes are equal, the alternative may be to use SHA512 hashes or something.